worker/durable-objects/PresenceRoom.ts

```
@@ -0,0 +1,122 @@
+        });
+      }
+      return new Response('expected websocket upgrade', { status: 426 });
+    }
+
+    const url = new URL(request.url);
+    const userId = (url.searchParams.get('userId') ?? '').trim();
+    if (!userId) {
+      return new Response('userId required', { status: 400 });
+    }
+
+    const pair = new WebSocketPair();
```

ESLint

[error] 'WebSocketPair' is not defined.

(no-undef)